pkg/operator/controller/ingress/deployment.go

```diff
@@ -660,6 +663,20 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 		env = append(env, corev1.EnvVar{Name: RouterHardStopAfterEnvName, Value: value})
 	}
 
+	// Only check for custom header buffer settings if
+	// ci.Spec.HTTPHeaderBuffer is not empty.
+	if (ci.Spec.HTTPHeaderBuffer != operatorv1.IngressControllerHTTPHeaderBuffer{}) {
```

Comment:
It seems like you could just pickup non-zero values and this check is not necessary.

Reply:
yea, we talked offline and I confirmed locally. Fixed.